contracts/modules/licensing/PILicenseTemplate.sol

```diff
@@ -0,0 +1,402 @@
+        address[] calldata originalIpIds,
+        uint256[] calldata licenseTermsIds,
+        address derivativeIpOwner
+    ) external override returns (bool) {
+        if (!_verifyCompatibleLicenseTerms(licenseTermsIds)) {
+            return false;
+        }
+        for (uint256 i = 0; i < licenseTermsIds.length; i++) {
+            if (!_verifyRegisterDerivative(derivativeIpId, originalIpIds[i], licenseTermsIds[i], derivativeIpOwner)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    function getRoyaltyPolicy(
+        uint256 licenseId
+    ) external view returns (address royaltyPolicy, bytes memory royaltyData, uint256 mintingFee, address currency) {
+        PILicenseTemplateStorage storage $ = _getPILicenseTemplateStorage();
+        PILTerms memory terms = $.licenseTerms[licenseId];
+        return (terms.royaltyPolicy, abi.encode(terms.commercialRevShare), terms.mintingFee, terms.currency);
+    }
+
+    function isTransferable(uint256 licenseId) external view override returns (bool) {
+        PILicenseTemplateStorage storage $ = _getPILicenseTemplateStorage();
+        return $.licenseTerms[licenseId].transferable;
+    }
+
+    function getEarlierExpireTime(
+        uint256 start,
+        uint256[] calldata licenseTermsIds
+    ) external view override returns (uint) {
+        if (licenseTermsIds.length == 0) {
+            return 0;
+        }
+        uint expireTime = _getExpireTime(start, licenseTermsIds[0]);
+        for (uint i = 1; i < licenseTermsIds.length; i++) {
+            uint newExpireTime = _getExpireTime(start, licenseTermsIds[i]);
+            if (newExpireTime < expireTime) {
+                expireTime = newExpireTime;
+            }
+        }
+        return expireTime;
+    }
+
+    function getExpireTime(uint256 start, uint256 licenseTermsId) external view returns (uint) {
+        return _getExpireTime(start, licenseTermsId);
+    }
+
+    function getLicenseTermsId(PILTerms calldata terms) external view returns (uint256 licenseTermsId) {
+        PILicenseTemplateStorage storage $ = _getPILicenseTemplateStorage();
+        bytes32 licenseTermsHash = keccak256(abi.encode(terms));
+        return $.hashedLicenseTerms[licenseTermsHash];
+    }
+
+    function totalRegisteredLicenseTerms() external view returns (uint256) {
+        PILicenseTemplateStorage storage $ = _getPILicenseTemplateStorage();
+        return $.licenseTermsCounter;
+    }
+
+    function supportsInterface(
+        bytes4 interfaceId
+    ) public view virtual override(BaseLicenseTemplate, IERC165) returns (bool) {
+        return interfaceId == type(IPILicenseTemplate).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    /// @notice Returns the stringified JSON policy data for the LicenseRegistry.uri(uint256) method.
+    /// @dev Must return OpenSea standard compliant metadata.
+    function toJson(uint256 licenseTermsId) public view returns (string memory) {
```

Comment:
why do we need this one?  LicenseTemplate is an NFT?

Reply:
It is for presenting the license into License Token's metadata returned by tokenURI.